src/dune_rules/artifact_substitution.ml

```diff
@@ -52,7 +52,7 @@ type conf =
   }
 
 let mac_codesign_hook ~codesign path =
-  Process.run Strict codesign [ "--sign"; "-"; Path.to_string path ]
+  Process.run Strict codesign [ "-s"; "-"; "-f"; Path.to_string path ]
```

Comment:
So these flags work with both tools?

Reply:
That's the goal. I tested this with Nix and it worked well. If someone has an easy way to verify that this works as expected, then we should be good to go.

Reply:
My expectation is that it will work, based on:

```shell
$ /usr/bin/codesign
Usage: codesign -s identity [-fv*] [-o flags] [-r reqs] [-i ident] path ... # sign
       codesign -v [-v*] [-R=<req string>|-R <req file path>] path|[+]pid ... # verify
       codesign -d [options] path ... # display contents
       codesign -h pid ... # display hosting paths
```

Reply:
I tested the patch on a "vanilla" M1 mac and it works, but...

I'm not sure we need `-f`: we only call `codesign` when substitutions invalidate the signature, so in the case there are none, the linker-provided signature should continue to be valid.

The reason I'm asking is that I'm not seeing failures in the test suite without `-f`, but adding it adds some extra logging output (so the case happens in the test suite) that we'd need to either suppress or postprocess in the case it's not the same with `darwin.sigtool`.

So if it works for your use case without `-f` I'd prefer not to include it.

Reply:
BTW, if you're including `codesign` in `PATH` for 3.5, I suppose we could revert #6231 too?

Reply:
(scratch that last part, the lazy detection seems more correct)

Reply:
```suggestion
  Process.run Strict codesign [ "-s"; "-"; Path.to_string path ]
```